examples/Receiver_Response.py

```diff
@@ -0,0 +1,164 @@
+# {{{ make spline for power going into RX box
+control.sort("nu_test")
+Pin_spline = control.spline_lambda()
+# }}}
+# }}}
+# {{{ Calculate $dg^2$
+for j, nodename in enumerate(file2_nodes):
+    d = psd.find_file(
+        file2,
+        exp_type=data_dir,
+        expno=nodename,
+    )
+    carrier = (
+        d.get_prop("acq_params")["carrierFreq_MHz"] * 1e6
+    )  # $\nu_{RX,LO}$
+    if j == 0:
+        # Allocate and array that's shaped like one of the datasets
+        # acquired (with 100 captures and a direct t dimension)
+        # but add axis to store the frequency of the test signal
+        rec_data = (d.shape + ("nu_test", len(nu_test_kHz))).alloc()
+        rec_data.setaxis("t", d.getaxis("t")).set_units("t", "s")
+        rec_data.setaxis("nu_test", nu_test_kHz).set_units("nu_test", "Hz")
+    rec_data["nu_test", j] = d
+    SW = str(d.get_prop("acq_params")["SW_kHz"]) # For labeling final plot
+rec_data.sort("nu_test")
+rec_data.rename("nScans", "capture")  # To be more consistent with the
+#                                        oscilloscope data rename the nScans
+#                                        dimension
+acq_time = np.diff(rec_data.getaxis("t")[np.r_[0, -1]]).item()
+# {{{ Calculate PSD for each frequency (we will calculate power from the A
+#     of the convolved test signal)
+rec_data.ft("t", shift=True)  # $dg\sqrt{s/Hz}$
+rec_data = abs(rec_data) ** 2  # $dg^{2}*s/Hz$
+rec_data.mean("capture")
+rec_data /= acq_time  # $dg^{2}/Hz$
+# Convolve using $\lambda_{G}$ specified above
+rec_data.convolve("t", lambda_G, enforce_causality=False)
+rec_data.ift("t")
+rec_data.run(np.conj)  # Empirically needed to give offset
+#                        that increases with field
+rec_data.ft("t")
+# }}}
+# {{{ Calculate power of test signal (Eq. S3)
+rec_data["t"] += carrier
+rec_data.run(np.max, "t")
+rec_data *= (lambda_G / (2 * np.sqrt(np.log(2)))) * np.sqrt(2 * np.pi)
+# }}}
+# }}}
+# {{{ Calculate receiver response as function of frequencies
+# Make axis of finely spaced frequencies to feed to spline
+rec_data["nu_test"] -= carrier # center data at 0 MHz
+Dnu = np.linspace(
+    (carrier) - (rec_data.getaxis("nu_test")[-1] / 2),
+    (carrier) + (rec_data.getaxis("nu_test")[-1] / 2),
+    len(rec_data.getaxis("nu_test")),
+)
+P_in = Pin_spline(Dnu)  # Generate spline of input powers
+dig_filter = rec_data / P_in
+# }}}
+# {{{ Fit receiver response
+A, omega, delta_nu, nu_test = symbols("A omega delta_nu nu_test", real=True)
+func_form = A * sp.sinc((2 * np.pi * (nu_test - omega)) / (delta_nu)) ** 2
+f = psd.lmfitdata(dig_filter)
+f.functional_form = func_form
+f.set_guess(
+    A=dict(value=1.6e17, min=1e17, max=9e19),
+    omega=dict(value=1, min=-100, max=100),
+    delta_nu=dict(value=9e3, min=3, max=1e4),
+)
+f.fit()
+fit = f.eval()
+# }}}
+with psd.figlist_var() as fl:
+    fl.next("Receiver Response")
+    fl.plot(for_plot(dig_filter), "o")
+    fl.plot(for_plot(fit), color="red", alpha=0.5)
```

Comment:
why are you not showing the image of the 2D response?

Reply:
fixed